blockchain/setup.go

```diff
@@ -1,210 +1,121 @@
-		"peerorg1Admin",
-	)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to get the organisation user failed: %v", err)
+	// Create channel
+	req := chmgmt.SaveChannelRequest{ChannelID: setup.ChannelID, ChannelConfig: setup.ChannelConfig + "chainhero.channel.tx", SigningIdentity: orgAdminUser}
+	if err = chMgmtClient.SaveChannel(req); err != nil {
+		return fmt.Errorf("failed to create channel: %v", err)
 	}
 
-	// Initialize the channel "mychannel" based on the genesis block by
-	//   1. locating in fixtures/channel/mychannel.tx and
-	//   2. joining the peer given in the configuration file to this channel
-	if err := fcutil.CreateAndJoinChannel(client, ordererUser, orgUser, channel, setup.ChannelConfig); err != nil {
-		return nil, fmt.Errorf("CreateAndJoinChannel return error: %v", err)
-	}
-
-	// Give the organisation user to the client for next proposal
-	client.SetUserContext(orgUser)
+	// Allow orderer to process channel creation
+	time.Sleep(time.Second * 5)
 
-	// Setup Event Hub
-	// This will allow us to listen for some event from the chaincode
-	// and act on it. We won't use it for now.
-	eventHub, err := getEventHub(client)
+	// Org resource management client
+	setup.admin, err = setup.sdk.NewClient(fabsdk.WithUser(setup.OrgAdmin)).ResourceMgmt()
 	if err != nil {
-		return nil, err
-	}
-	if err := eventHub.Connect(); err != nil {
-		return nil, fmt.Errorf("Failed eventHub.Connect() [%s]", err)
+		return fmt.Errorf("failed to create new resource management client: %v", err)
 	}
-	setup.EventHub = eventHub
 
-	// Tell that the initialization is done
-	setup.Initialized = true
+	// Org peers join channel
+	if err = setup.admin.JoinChannel(setup.ChannelID); err != nil {
+		return fmt.Errorf("org peers failed to join the channel: %v", err)
+	}
 
-	return &setup, nil
+	fmt.Println("Initialization Successful")
+	return nil
 }
 
-// getEventHub initialize the event hub
-func getEventHub(client api.FabricClient) (api.EventHub, error) {
-	eventHub, err := events.NewEventHub(client)
-	if err != nil {
-		return nil, fmt.Errorf("Error creating new event hub: %v", err)
-	}
-	foundEventHub := false
-	peerConfig, err := client.GetConfig().GetPeersConfig()
+func (setup *FabricSetup) InstallAndInstantiateCC() error {
+
+	// Create chaincode package for our chaincode
+	ccPkg, err := packager.NewCCPackage(setup.ChaincodePath, setup.ChaincodeGoPath)
 	if err != nil {
-		return nil, fmt.Errorf("Error reading peer config: %v", err)
-	}
-	for _, p := range peerConfig {
-		if p.EventHost != "" && p.EventPort != 0 {
-			fmt.Printf("EventHub connect to peer (%s:%d)\n", p.EventHost, p.EventPort)
-			eventHub.SetPeerAddr(fmt.Sprintf("%s:%d", p.EventHost, p.EventPort),
-				p.TLS.Certificate, p.TLS.ServerHostOverride)
-			foundEventHub = true
-			break
-		}
+		return fmt.Errorf("failed to create chaincode package: %v", err)
 	}
 
-	if !foundEventHub {
-		return nil, fmt.Errorf("No EventHub configuration found")
+	// Install our chaincode on org peers
```

Comment:
Explain in more detail. Tell what is the install process (regarding instantiate process).

Reply:
@MaximeAubanel

Reply:
Done